test/league_stats_test.rb

```diff
@@ -0,0 +1,166 @@
+require './lib/game'
+require './lib/game_stats'
+require './lib/league_stats'
+require 'pry'
+
+class LeagueStatsTest < Minitest::Test
+
+  def setup
+    #team_1_info = {team_id: "1", franchiseId: "23", shortName: "New Jersey", teamName: "Devils", abbreviation: "NJD", link: "/api/v1/teams/1"}
+    team_3_info = {team_id: "3", franchiseId: "23", shortName: "", teamName: "Rangers", abbreviation: "NJD", link: "/api/v1/teams/1"}
+    #team_4_info = {team_id: "4", franchiseId: "16", shortName: "Philadelphia", teamName: "Flyers", abbreviation: "PHI", link: "/api/v1/teams/4"}
+    team_6_info = {team_id: "6", franchiseId: "16", shortName: "", teamName: "Bruins", abbreviation: "PHI", link: "/api/v1/teams/4"}
+    team_10_info = {team_id: "10", franchiseId: "16", shortName: "", teamName: "Maple Leafs", abbreviation: "PHI", link: "/api/v1/teams/4"}
+    team_12_info = {team_id: "12", franchiseId: "16", shortName: "", teamName: "Hurricanes", abbreviation: "PHI", link: "/api/v1/teams/4"}
+    team_13_info = {team_id: "13", franchiseId: "16", shortName: "", teamName: "Panthers", abbreviation: "PHI", link: "/api/v1/teams/4"}
+    team_14_info = {team_id: "14", franchiseId: "16", shortName: "", teamName: "Lightning", abbreviation: "PHI", link: "/api/v1/teams/4"}
+    team_23_info = {team_id: "23", franchiseId: "16", shortName: "", teamName: "Canucks", abbreviation: "PHI", link: "/api/v1/teams/4"}
+    team_52_info = {team_id: "52", franchiseId: "16", shortName: "", teamName: "Jets", abbreviation: "PHI", link: "/api/v1/teams/4"}
+
+    #@team_1 = Team.new(team_1_info)
+    @team_3 = Team.new(team_3_info)
+    #@team_4 = Team.new(team_4_info)
+    @team_6 = Team.new(team_6_info)
+    @team_10 = Team.new(team_10_info)
+    @team_12 = Team.new(team_12_info)
+    @team_13 = Team.new(team_13_info)
+    @team_14 = Team.new(team_14_info)
+    @team_23 = Team.new(team_23_info)
+    @team_52 = Team.new(team_52_info)
+    @teams = [
+              #@team_1,
+              @team_3,
+              #@team_4,
+              @team_6,
+              @team_10,
+              @team_12,
+              @team_13,
+              @team_14,
+              @team_23,
+              @team_52
+             ]
+
+    game_1_info = {:game_id=>"2012030221", :season=>"20122013", :type=>"P", :date_time=>"2013-05-16", :away_team_id=>"3", :home_team_id=>"6", :away_goals=>"2", :home_goals=>"3", :outcome=>"home win OT", :home_rink_side_start=>"left", :venue=>"TD Garden", :venue_link=>"/api/v1/venues/		null", :venue_time_zone_id=>"America New_York", :venue_time_zone_offset=>"-4", :venue_time_zone_tz=>"EDT"}
+    game_2_info = {:game_id=>"2012030222", :season=>"20122013", :type=>"P", :date_time=>"2013-05-19", :away_team_id=>"3", :home_team_id=>"6", :away_goals=>"2", :home_goals=>"5", :outcome=>"home win REG", :home_rink_side_start=>"left", :venue=>"TD Garden", :venue_link=>"/api/v1/venues/		null", :venue_time_zone_id=>"America New_York", :venue_time_zone_offset=>"-4", :venue_time_zone_tz=>"EDT"}
```

Comment:
Several of the teams in these games don't play as both home and away, causing the error I was receiving.

Reply:
> Several of the teams in these games don't play as both home and away, causing the error I was receiving.

I fixed that issue and all of the "ArgumentError: comparison of Float with NaN failed" issues. All the teams should play the necessary games now so I don't know what yours is missing.

Reply:
From what I could tell it was from the attempts to divide a Float by NaN when a game didn't play either a home or away game and their stats remained at zero. I got it running by trimming down the games and teams to two teams playing three games (with one where the home/away arrangement is switched).

Reply:
Like I said, I added the necessary games already and it is not an issue in my version of the code. I've added a lot of stuff that we should look at before trimming anything down. Wanted more teams to test all the methods. This work is getting to be a lot and should be merged to master soon (unless you still have major issues). I think we're gonna waste time doubling up work if you are making edits to make it work when I have already made other edits and it is working. This is turning less into a "feature" the more edits we keep making without merging. Also this push can basically be abandoned. Start looking at the "season stats" branch I pushed. It encompasses all the changes discussed here, as well as a lot of shifting methods around.